plugins/net_plugin/net_plugin.cpp

```diff
@@ -254,8 +256,9 @@ namespace eosio {
       bool                                  use_socket_read_watermark = false;
       /** @} */
 
-      mutable std::shared_mutex             connections_mtx;
+      mutable std::shared_mutex             connections_mtx; // protects both connections and security_group
```

Comment:
As implemented `connections_mtx` does not protect the `security_group`. It looks to me like `security_group` is only accessed and modified via the application thread.

Reply:
It will need to be checked by the connection itself after the certificates are exchanged, to determine if the incoming participant is in the security group and can receive blocks or not.  I thought it would be simpler for it to be covered by the connections mutex rather than having a separate one and having to make sure that they are taken in the correct order.

Reply:
I don't think it needs a mutex as it is only accessed via the application thread, right?

Reply:
The TLS connection portion isn't implemented yet. The connection will have an extra handshake where the participant will be identified and the connection will then have to search in the security group to see if the participant is in or out of the group, and that won't be on the application thread, it will be on one of the net_plugin thread pool threads.

Reply:
In that case it needs to be a lock_guard not a shared_lock of the connections_mtx.

Reply:
Changed to lock_guard